dannce/engine/generator.py

```diff
@@ -179,10 +186,12 @@ def load_vid_frame(
 
             # This deals with a strange indexing error in the pup data.
             try:
-                im = vid.get_data(frame_num).astype("uint8")
+                # im = vid.get_data(frame_num).astype("uint8")
+                im = vid.get_frame(frame_num)
             except IndexError:
                 print("Indexing error using previous frame")
-                im = vid.get_data(frame_num - 1).astype("uint8")
+                # im = vid.get_data(frame_num - 1).astype("uint8")
+                im = vid.get_frame(frame_num)
```

Comment:
I wrote this catch to deal with an annoying bug in Opcon's recordings where the last frame throws an indexerror despite the file having the correct number of indices. The recordings are fast enough that it basically doesn't matter and I haven't seen this happen with any other data, including anything from campy.  

So far I've only ever seen this happen with imageio. I took off the -1 to actually be able to see if the error was happening in opencv, and it doesn't seem to happen, since its never been thrown. Adding it back could make sense for us down the line when using imageio.

Reply:
ok, will leave this to fix when I add the imageio flag to next PR